app/src/main/java/cm/aptoide/pt/social/commentslist/PostCommentsService.java

```diff
@@ -0,0 +1,81 @@
+package cm.aptoide.pt.social.commentslist;
+
+import android.content.SharedPreferences;
+import cm.aptoide.pt.dataprovider.interfaces.TokenInvalidator;
+import cm.aptoide.pt.dataprovider.model.v7.Comment;
+import cm.aptoide.pt.dataprovider.ws.BodyInterceptor;
+import cm.aptoide.pt.dataprovider.ws.v7.BaseBody;
+import cm.aptoide.pt.dataprovider.ws.v7.ListCommentsRequest;
+import java.util.Collections;
+import java.util.List;
+import okhttp3.OkHttpClient;
+import retrofit2.Converter;
+import rx.Single;
+
+/**
+ * Created by jdandrade on 28/09/2017.
+ */
+
+class PostCommentsService {
+
+  private final int limit;
+  private final int initialOffset;
+  private final BodyInterceptor<BaseBody> bodyInterceptor;
+  private final OkHttpClient httpClient;
+  private final Converter.Factory converterFactory;
+  private final TokenInvalidator tokenInvalidator;
+  private final SharedPreferences sharedPreferences;
+  private int currentOffset;
+  private int total;
+  private boolean loading;
+
+  PostCommentsService(int limit, int initialOffset, int initialTotal,
```

Comment:
int limit, int initialOffset and int initialTotal should be handled by repository

Reply:
I agree, we already talked about this, I suggest that I do this in parallel with what @paquiquinho is missing. If he finishes first, close this PR (if everything else is ok) and let me know so I open a ticket.